src/app/api.ts

```diff
@@ -1075,6 +1104,99 @@ export function newAPI(): API {
     return secrets;
   }
 
+  async function createSecretProjects(
+    projectref: string,
+    secretname: string,
+    secretValue: SecretValues,
+    signal?: AbortSignal
+  ): Promise<SecretResponse> {
+    const apiURL = baseURL();
+    apiURL.pathname +=
+      '/projects/' + encodeURIComponent(projectref) + '/secrets';
+    const init = {
+      method: 'POST',
+      body: JSON.stringify({
+        name: secretname,
+        type: SecretType.Internal,
+        data: secretValue,
+      }),
+      signal,
+    };
+
+    const res = await fetch(apiURL.toString(), init);
+
+    const secret = TypedJSON.parse(await res.text(), SecretResponse);
+    if (!secret) throw new ApiError();
+
+    return secret;
+  }
+
+  async function createSecretProjectGroups(
+    projectgroupref: string,
+    secretname: string,
+    secretValue: SecretValues,
+    signal?: AbortSignal
+  ): Promise<SecretResponse> {
+    const apiURL = baseURL();
+    apiURL.pathname +=
+      '/projectgroups/' + encodeURIComponent(projectgroupref) + '/secrets';
+
+    const init = {
+      method: 'POST',
+      body: JSON.stringify({
+        name: secretname,
+        type: SecretType.Internal,
+        data: secretValue,
+      }),
+      signal,
+    };
+
+    const res = await fetch(apiURL.toString(), init);
+
+    const secret = TypedJSON.parse(await res.text(), SecretResponse);
+    if (!secret) throw new ApiError();
+
+    return secret;
+  }
+
+  async function deleteSecretProjectGroups(
+    secretname: string,
+    username: string,
+    signal?: AbortSignal
+  ): Promise<void> {
+    const apiURL = baseURL();
+    apiURL.pathname +=
+      '/projectgroups/user%2F' + username + '/secrets/' + secretname;
```

Comment:
The same comments of the previous reviews (like https://github.com/agola-io/agola-web/pull/89#discussion_r1325706762 and https://github.com/agola-io/agola-web/pull/89#discussion_r1325707999) also apply to these and other methods.